pycloudlib/gce/tests/test_cloud.py

```diff
@@ -0,0 +1,148 @@
+"""Tests related to pycloudlib.gce.cloud module."""
+import mock
+import pytest
+
+from pycloudlib.cloud import ImageType
+from pycloudlib.gce.cloud import GCE
+
+# mock module path
+MPATH = "pycloudlib.gce.cloud."
+
+
+class FakeGCE(GCE):
+    """GCE Class that doesn't load config or make requests during __init__."""
+
+    # pylint: disable=super-init-not-called
+    def __init__(self, *_, **__):
+        """Fake __init__ that sets mocks for needed variables."""
+        self._log = mock.MagicMock()
+        self.compute = mock.MagicMock()
+
+
+# pylint: disable=protected-access,missing-function-docstring
+class TestGCE:
+    """General GCE testing."""
+
+    @pytest.mark.parametrize(
+        [
+            "release",
+            "arch",
+            "api_side_effects",
+            "expected_filter_calls",
+            "expected_image_list",
+        ],
+        [
+            ("xenial", "arm64", Exception(), [], []),
```

Comment:
This is a really good start to instrumenting coverage for GCE. Thanks for all the parametrized tests.
One thing I think we generally want to do for test failure readability in the future is using `pytest.param(...., id=unique_descriptive_name)` around large sets of parameters to explain the intent of the test. With a descriptive id, we see better failure messages instead of an abbreviated dict representation of the parametrized values.

Reply:
Here's a consolidated diff suggestion that I'll apply here. Given it's non-functional and trivial, I'll approve and merge this
```diff
diff --git a/pycloudlib/gce/tests/test_cloud.py b/pycloudlib/gce/tests/test_cloud.py
index 990a640..3669843 100644
--- a/pycloudlib/gce/tests/test_cloud.py
+++ b/pycloudlib/gce/tests/test_cloud.py
@@ -32,8 +32,15 @@ class TestGCE:
             "expected_image_list",
         ],
         [
-            ("xenial", "arm64", Exception(), [], []),
-            (
+            pytest.param(
+                "xenial",
+                "arm64",
+                Exception(),
+                [],
+                [],
+                id="xenial_no_arm64_support_zero_sdk_list_calls",
+            ),
+            pytest.param(
                 "xenial",
                 "x86_64",
                 [{"items": [1, 2, 3]}, Exception()],
@@ -46,8 +53,9 @@ class TestGCE:
                     )
                 ],
                 [1, 2, 3],
+                id="xenial_x86_64_suppport_one_sdk_list_call_empty_pagetoken",
             ),
-            (
+            pytest.param(
                 "kinetic",
                 "arm64",
                 [
@@ -70,6 +78,7 @@ class TestGCE:
                     ),
                 ],
                 [1, 2, 3, 4, 5, 6],
+                id="non_xenial_arm64_suppport_one_sdk_list_call_per_page",
             ),
         ],
     )
```

Reply:
TIL about `pytest.param` - I'll start using that everywhere I pytest now, thanks!!